signer/src/storage/postgres.rs

```diff
@@ -1951,6 +2062,88 @@ impl super::DbWrite for PgStore {
 
         Ok(())
     }
+
+    async fn write_sweep_transaction(
+        &self,
+        transaction: &model::SweepTransaction,
+    ) -> Result<(), Error> {
+        // We're doing multiple inserts here so we wrap them in a transaction.
+        let mut tx = self.0.begin().await.map_err(Error::SqlxBeginTransaction)?;
+
+        sqlx::query(
+            "
+            INSERT INTO sweep_transactions (
+                txid
+              , signer_prevout_txid
+              , signer_prevout_output_index
+              , signer_prevout_amount
+              , signer_prevout_script_pubkey
+              , amount
+              , fee
+              , created_at_block_hash
+              , market_fee_rate
+            ) 
+            VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9)",
+        )
+        .bind(transaction.txid)
+        .bind(transaction.signer_prevout_txid)
+        .bind(transaction.signer_prevout_output_index as i32)
+        .bind(transaction.signer_prevout_amount as i64)
+        .bind(transaction.signer_prevout_script_pubkey.clone())
+        .bind(transaction.amount as i64)
+        .bind(transaction.fee as i64)
+        .bind(transaction.created_at_block_hash)
+        .bind(transaction.market_fee_rate)
+        .execute(&mut *tx)
+        .await
+        .map_err(Error::SqlxQuery)?;
+
+        // Insert the swept deposits.
+        for deposit in transaction.swept_deposits.iter() {
+            sqlx::query(
+                "
+                INSERT INTO swept_deposits (
+                    sweep_transaction_txid
+                  , input_index
+                  , deposit_request_txid
+                  , deposit_request_output_index
+                )
+                VALUES ($1, $2, $3, $4)",
```

Comment:
We might want to add an "ON CONFLICT DO NOTHING" to the end of these queries.

Reply:
Added to all three in [d181170](https://github.com/stacks-network/sbtc/pull/710/commits/d1811705b06b1167b0e178f41a92f81e67242b14)

Reply:
Just a side note, I'm generally not overly fond of using "ON CONFLICT DO NOTHING" because it can mask potential issues; like, it could be that we've somehow ended up with two entries with the same PK but different data, and this will blindly ignore what maybe should be an update or indicative of a problem elsewhere.

Reply:
Good point and we should aim to remove them later. This is a bandaid for the fact that we bail at the first sign of trouble.

Reply:
Actually, maybe that's not the best reason to add them. I originally wanted to add them because I was more worried about duplicate messages than messages that have different data but matching primary keys. I still feel that way. The bail early thing is a different beast. I barely considered the different data stuff issue.